modules/aws/cloudfront-s3/variables.tf

```diff
@@ -0,0 +1,171 @@
+}
+
+/* CF Origin Bucket */
+
+variable "cf_origin_create_bucket" {
+  description = "Whether or not create cf origin bucket. If say 'true' bucket would be created with the name what ever you give in 'cf_origin_bucket_name'"
+  type        = bool
+}
+
+variable "cf_origin_bucket_name" {
+  description = "Name for cf origin bucket to while creating or Pass an existing bucket name when 'cf_origin_create_bucket' is set to false"
+  type        = string
+}
+
+variable "attach_policy" {
+  description = "If your existing bucket has bucket policy and you want to append cf policy to s3 bucket say 'trre' else 'false'"
+  type        = bool
+  default     = false
+}
+
+variable "block_public_acls" {
+  description = "Whether Amazon S3 should block public ACLs for this bucket."
+  type        = bool
+  default     = true
+}
+
+variable "block_public_policy" {
+  description = "Whether Amazon S3 should block public bucket policies for this bucket."
+  type        = bool
+  default     = true
+}
+
+variable "ignore_public_acls" {
+  description = "Whether Amazon S3 should ignore public ACLs for this bucket."
+  type        = bool
+  default     = true
+}
+
+variable "default_root_object" {
+  description = "The object that you want CloudFront to return (for example, index.html) when an end user requests the root URL."
+  type        = string
+  default     = "index.html"
+}
+
+variable "assets_dir_path" {
+  description = "Directory path where assets would be stored. The same path would be cached by cloudfront."
+  default = null
+  type        = string
+}
+
+
+variable "is_ipv6_enabled" {
+  description = "Whether the IPv6 is enabled for the distribution."
+  type        = bool
+  default     = true
+}
+
+variable "price_class" {
+  description = "The price class for this distribution. One of PriceClass_All, PriceClass_200, PriceClass_100"
+  type        = string
+  default     = "PriceClass_All"
+}
+
+variable "retain_on_delete" {
+  description = "Disables the distribution instead of deleting it when destroying the resource through Terraform. If this is set, the distribution needs to be deleted manually afterwards."
+  type        = bool
+  default     = false
+}
+
+variable "wait_for_deployment" {
+  description = "If enabled, the resource will wait for the distribution status to change from InProgress to Deployed. Setting this tofalse will skip the process."
+  type        = bool
+  default     = true
+}
+
+variable "origin_protocol_policy" {
+  description = "Protocol that cloudfront to use when connecting to the origin. Supported values (http-only, https-only, or match-viewer)"
+  type        = string
+  default = "https-only"
+}
+
+##Cache Behavior Arguments###
+variable "allowed_methods" {
+  description = "Controls which HTTP methods CloudFront processes and forwards to your custom origin. for ex: 'DELETE', 'GET', 'HEAD', 'OPTIONS', 'PATCH', 'POST', 'PUT'"
+  type        = list(any)
+  default = ["GET", "HEAD"]
```

Comment:
should support all the above by default no?

Reply:
Added "Options" as well